### PR TITLE
[Validator] Added docs about the format option

### PR DESCRIPTION
**Pull request by dosten**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes, see symfony/symfony#17553 |
| Applies to | 3.1+ |
| Fixed tickets | #6313 |

### CODE REVIEW

**Comment by dosten** [comment]
ping @xabbuh

File: reference/constraints/DateTime.rst [hunk 1]
```diff
@@ -85,4 +86,16 @@ message
...
 
+format
+~~~~~~
+
+**type**: ``string`` **default**: ``Y-m-d H:i:s``
+
+.. versionadded:: 3.1
+    The ``format`` option was introduced in Symfony 3.1.
+
+This option allows to validate a custom date format. See `date_create_from_format`_ for formatting options.
```
Comment (xabbuh) [inline reference/constraints/DateTime.rst]: I would change the label to `DateTime:createFromFormat()`:

```rst
See `DateTime::createFromFormat <date_create_from_format>`_ for [...]
```

**Comment by dosten** [comment]
@xabbuh comments addressed

File: reference/constraints/DateTime.rst [hunk 2]
```diff
@@ -78,6 +79,16 @@ Basic Usage
...
 
+format
+~~~~~~
+
+**type**: ``string`` **default**: ``Y-m-d H:i:s``
+
+.. versionadded:: 3.1
+    The ``format`` option was introduced in Symfony 3.1.
+
+This option allows to validate a custom date format. See `DateTime::createFromFormat <date_create_from_format>`_ for formatting options.
```
Comment (xabbuh) [inline reference/constraints/DateTime.rst]: Can you please the line after the first word that crosses the 72nd character?

**Comment by xabbuh** [comment]
:+1: I just left some very minor comment.

**Comment by dosten** [comment]
@xabbuh done

**Comment by xabbuh** [comment]
Great! :)

**Comment by WouterJ** [comment]
Thanks for the docs @dosten!

Fyi, I've changed the external link to use the custom `:phpmethod:` role in 6e73637.

**Pull request merged**